library/leaderboard_fetch.js

```diff
@@ -8,152 +8,201 @@ module.exports = (client, db) => {
      * this entire file in the upcoming days
     */
 
+  client.findCurrentPrackers = (guild, callback) => {
+    // playtimes only get updated when a user leaves/mutes a channel. Therefore, in order to keep up-to-date statistics,
+    // find out what users are currently in permitted voice channels, then add their times as if current.
+    client.loadGuildData(guild, res => {
+      let currentPrackers = new Map()
+      res.permitted_channels.forEach(channel => {
+        let vc = client.guilds.get(guild).channels.get(channel)
+        if (vc != null) {
+          vc.members.forEach(member => {
+            // these conditions should be equivalent but maybe they were already pracking when the bot came up.
```

Comment:
Right - current behaviour can't track practise sessions that started before the bot came up. (The workaround for now is simply for a mod to mute everybody while redeploying the bot.) The live lb keeps the behaviour consistent - if the bot *would* award you time when you unmute, it will now update the time as you're live. If it wouldn't, then it won't update the live time.

One convenient side effect of this is that now you can tell ahead of time if the bot is tracking you, instead of finding out that the bot wasn't tracking you only after you finish a session.